internal/provider/fwprovider/intercept.go

```diff
@@ -243,7 +245,104 @@ type tagsDataSourceInterceptor struct {
 }
 
 func (r tagsDataSourceInterceptor) read(ctx context.Context, request datasource.ReadRequest, response *datasource.ReadResponse, meta *conns.AWSClient, when when, diags diag.Diagnostics) (context.Context, diag.Diagnostics) {
-	// TODO
+	if r.tags == nil {
```

Comment:
Not as part of this PR, but we should try and share this code with the equivalent for resources.

Reply:
Absolutely, there's a lot of overlap with this and the SDK versions, too